engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -4026,25 +4027,39 @@ public NicProfile addVmToNetwork(final VirtualMachine vm, final Network network,
 
             final Object jobException = _jobMgr.unmarshallResultObject(outcome.getJob());
             if (jobException != null) {
-                if (jobException instanceof ResourceUnavailableException) {
-                    throw (ResourceUnavailableException)jobException;
-                } else if (jobException instanceof ConcurrentOperationException) {
-                    throw (ConcurrentOperationException)jobException;
-                } else if (jobException instanceof InsufficientCapacityException) {
-                    throw (InsufficientCapacityException)jobException;
-                } else if (jobException instanceof RuntimeException) {
-                    throw (RuntimeException)jobException;
-                } else if (jobException instanceof Throwable) {
-                    throw new RuntimeException("Unexpected exception", (Throwable)jobException);
-                } else if (jobException instanceof NicProfile) {
-                    return (NicProfile)jobException;
-                }
+                return getNewNicProfile(vm, network, requested, outcome, jobException);
             }
 
             throw new RuntimeException("Unexpected job execution result");
         }
     }
 
+    @NotNull
+    private NicProfile getNewNicProfile(VirtualMachine vm, Network network, NicProfile requested, Outcome<VirtualMachine> outcome, Object jobException) throws ResourceUnavailableException, InsufficientCapacityException {
+        if (s_logger.isDebugEnabled()) {
+            s_logger.debug(String.format("remove vmworkjob record %d for adding nic %s to net %s in vm %s",
+                    outcome.getJob().getId(),
+                    requested.getUuid(),
+                    network.getUuid(),
+                    vm.getUuid()));// remove vmworkjob record
+        }
+        _workJobDao.expunge(outcome.getJob().getId());
+
+        if (jobException instanceof ResourceUnavailableException) {
+            throw (ResourceUnavailableException) jobException;
+        } else if (jobException instanceof ConcurrentOperationException) {
+            throw (ConcurrentOperationException) jobException;
+        } else if (jobException instanceof InsufficientCapacityException) {
+            throw (InsufficientCapacityException) jobException;
+        } else if (jobException instanceof RuntimeException) {
+            throw (RuntimeException) jobException;
+        } else if (jobException instanceof Throwable) {
+            throw new RuntimeException("Unexpected exception", (Throwable) jobException);
+        } else if (jobException instanceof NicProfile) {
+            return (NicProfile) jobException;
+        }
```

Comment:
Should there be a default return if none of the cases are matched?

Reply:
there should actually be a throw. let me check